x-pack/legacy/plugins/fleet/server/libs/agent.ts

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import moment from 'moment';
+import {
+  AgentAdapter,
+  Agent,
+  SortOptions,
+  NewAgent,
+  AgentType,
+} from './adapters/agent/adapter_type';
+import { TokenLib } from './token';
+
+export class AgentLib {
+  constructor(private readonly agentAdater: AgentAdapter, private readonly tokens: TokenLib) {}
+
+  /**
+   * Enroll a new token into elastic fleet
+   */
+  public async enroll(
+    token: any,
+    type: AgentType,
+    config: { id: string; sharedId: string },
```

Comment:
If no configuration is provided, then this defaults to an I’d and a sharedId of `default` but if one is provided they both must be.
The shared ID passed in here is for the configuration, but it looks like your looking up other agents by the shared ID?

Reply:
I have to `sharedId` one `config. sharedId` for the config and `sharedId` for the agent

Reply:
Instead of having a `default` config, what do you think to allow to start with empty config just a name? it will made everything a lot easier to manager? 
And if you look to all the similar tools where you enroll agents, (new relic, datadog, ...) you always start by creating an application (configuration in our case), enroll, and after you can tune your config.

Reply:
Can we support both? Default and custom with just name?

Reply:
Yes we can support both, but for this lib I think it will change nothing you need to call it with a config even if it's the default config